pkg/scheduler/framework/runtime/framework.go

```diff
@@ -561,33 +561,34 @@ func (f *frameworkImpl) QueueSortFunc() framework.LessFunc {
 // *Status and its code is set to non-success if any of the plugins returns
 // anything but Success. If a non-success status is returned, then the scheduling
 // cycle is aborted.
-func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod) (status *framework.Status) {
+func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod) (_ *framework.PreFilterResult, status *framework.Status) {
 	startTime := time.Now()
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(preFilter, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
+	result := framework.PreFilterResult{}
 	for _, pl := range f.preFilterPlugins {
-		status = f.runPreFilterPlugin(ctx, pl, state, pod)
-		if !status.IsSuccess() {
-			status.SetFailedPlugin(pl.Name())
-			if status.IsUnschedulable() {
-				return status
+		r, s := f.runPreFilterPlugin(ctx, pl, state, pod)
+		if !s.IsSuccess() {
+			s.SetFailedPlugin(pl.Name())
+			if s.IsUnschedulable() {
+				return nil, s
 			}
-			return framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", pl.Name(), status.AsError())).WithFailedPlugin(pl.Name())
+			return nil, framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", pl.Name(), status.AsError())).WithFailedPlugin(pl.Name())
 		}
+		result.Merge(r)
```

Comment:
If some preFilter like `Fit` returns PreFilterResult as nil. then no matter what the other preFilters return. The result is always allNodes, right? Shouldn't we take the intersection here？

Reply:
Oh I find this in merge. We have special handling.
```go
	if p.NodeNames == nil && len(in.NodeNames) != 0 {
		p.NodeNames = sets.NewString()
	}
```
But there is still something strange. If A returns node1 and node2; B returns node2 and node3;
It seems like the result = (B ∪ C) = node1 node2 node3; Here only node2 meets the requirements for both A and B, Why not return node2 directly?